source/codegen/templates/service_helpers.mako

```diff
@@ -218,8 +218,17 @@ one_of_case_prefix = f'{namespace_prefix}{function_name}Request::{PascalFieldNam
 %>\
 %     if common_helpers.is_struct(parameter) or underlying_param_type == 'ViBoolean':
       std::vector<${underlying_param_type}> ${parameter_name}(${size}, ${underlying_param_type}());
-%     elif service_helpers.is_string_arg(parameter):
+## byte arrays are leveraging a string as a simple buffer wrapper
+%     elif service_helpers.is_string_arg(parameter) and parameter['type'] == 'ViInt8[]':
       std::string ${parameter_name}(${size}, '\0');
+## real strings leverages the underlying string buffer's null terminator as part of the size
```

Comment:
Are we confident this is the case for all drivers?

Reply:
This (poorly worded - I'll fix!) comment is referring to the code we're generating, not the behavior of the driver. The drivers definitely require that the buffer be big enough to hold the null-terminator. You can search ni-code for `sizeof('\0')` to see lots of examples!